src/server/worker/datum/iterator_old.go

```diff
@@ -0,0 +1,550 @@
+package datum
+
+//// Iterator is an interface which allows you to iterate through the datums
```

Comment:
It looks like some of this is implemented in `iterator.go` I would prefer deleting the code that has been ported, and only leaving comments for the unported stuff.